### ISSUE TITLE
Avoid deprecation warnings

### ISSUE THREAD

**Issue by yguclu** [pyccel/psydac#353]
NumPy warning:
> DeprecationWarning: `product` is deprecated as of NumPy 1.25.0, and will be removed in NumPy 2.0. Please use `prod` instead.

SymPy warning:
> SymPyDeprecationWarning: 
>  
>  non-Expr objects in a Matrix has been deprecated since SymPy 1.9. Use
>  list of lists, TableForm or some other data structure instead. See
>  https://github.com/sympy/sympy/issues/21497 for more info.

**Comment by kvrigor** [comment]
With #355 merged, should we close this issue?

**Comment by yguclu** [comment]
> With #355 merged, should we close this issue?

No, we cannot, because #355 did not address the most serious warning, which is the `ComplexWarning` raised by NumPy.
